builtin/providers/aws/resource_aws_ssm_document.go

```diff
@@ -0,0 +1,309 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ssm"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsSsmDocument() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsSsmDocumentCreate,
+		Read:   resourceAwsSsmDocumentRead,
+		Update: resourceAwsSsmDocumentUpdate,
+		Delete: resourceAwsSsmDocumentDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
```

Comment:
We don't need the `&schema.Schema` here - though that is currently being spread across the code base. For new resources it's preferable to use the Go 1.5+ style of not declaring the struct names in these lists as it results in cleaner syntax.